api/store.js

```diff
@@ -0,0 +1,176 @@
+const logger = require('./logger')('blob store');
+const fs = require('fs');
+const stream = require('stream');
+const fsBlobStore = require('fs-blob-store');
+
+const isReadableStream = obj => obj instanceof stream.Readable;
+
+// Creates a filestream blob store that reads/writes to
+// /dev/null.  Can't use the fsBlobStore for this because
+// it tries to create /dev/null and...  well, that doesn't
+// fails, as we might reasonably expect.
+const nullBlobStore = () => ({
+  createReadStream: () => fs.createReadStream('/dev/null'),
+  createWriteStream: () => fs.createWriteStream('/dev/null'),
+  exists: cb => {
+    cb(null, true);
+  },
+  remove: (key, cb) => {
+    cb(null);
+  }
+});
+
+/**
+ * Gets blob store based on configuration.
+ * @param {String} type - Blob store type
+ * @param {String} path - Path info for the blob store
+ * @returns Abstract blob store: https://github.com/maxogden/abstract-blob-store
+ */
+const getBlobStore = (type, path) => {
+  switch (type.toLowerCase()) {
+    case 'fs':
+      logger.silly('creating fs blob store');
+      return fsBlobStore(path);
+    default:
+      logger.silly('creating null blob store');
+      return nullBlobStore();
+  }
+};
+
+logger.silly('setting up...');
+// const store = getBlobStore(process.env.STORE_TYPE, process.env.STORE_PATH);
+
+class Store {
+  constructor(store) {
+    this.store = store;
+  }
+
+  /**
+   * @async
+   * Checks if a blob exists
+   * @param {String} key - Blob identifier
+   * @returns True if it exists, false if it doesn't.
+   */
+  async exists(key) {
+    return new Promise((resolve, reject) => {
+      logger.silly(`checking if blob exists [${key}]`);
+      this.store.exists(key, (err, exists) => {
+        if (err) {
+          logger.error(err);
+          reject(err);
+        } else {
+          logger.silly(`blob [${key}] does ${exists ? '' : 'not '}exist`);
+          resolve(exists);
+        }
+      });
+    });
+  }
+
+  /**
+   * @async
+   * Gets the data from a specified blob.
+   * @param {String} key - Blob identifier
+   * @returns {Buffer} A buffer of the data in the blob.
+   */
+  async get(key) {
```

Comment:
Not sure this method is actually useful, but it was helpful for testing.  My ***guess*** is that in practice, we'll want to get the read stream and pipe that directly to an HTTP response rather than reading it into memory.